src/i_video.c

```diff
@@ -1389,15 +1399,53 @@ static void CreateUpscaledTexture(boolean force)
     SDL_SetTextureScaleMode(texture_upscaled, SDL_ScaleModeLinear);
 }
 
+static void UpdateViewport(void)
+{
+    int w, h;
+    SDL_GetRendererOutputSize(renderer, &w, &h);
+
+    double real_aspect = (double)w / h;
+    double want_aspect = CurrentAspectRatio();
+
+    float scalex = (float)w / video.width;
+    float scaley = (float)h / actualheight;
+
+    // Clear the scale because we're setting viewport in output coordinates
+    SDL_RenderSetScale(renderer, 1.0f, 1.0f);
+
+    SDL_Rect viewport = {0};
+
+    if (fabs(want_aspect - real_aspect) < 0.0001)
+    {
+        viewport.w = w;
+        viewport.h = h;
+        letterboxed = false;
+    }
+    else if (want_aspect > real_aspect)
+    {
+        viewport.w = w;
+        viewport.h = (int)floor(actualheight * scalex);
+        viewport.y = (h - viewport.h) / 2;
+        letterboxed = true;
```

Comment:
```c
scaley = scalex;
```

Reply:
This is from the SDL code, see https://github.com/libsdl-org/SDL/blob/a3546e50697f2424a39f991148708c60903c553b/src/render/SDL_render.c#L2391

Reply:
The scale must be uniform here. It's the same in SDL: https://github.com/libsdl-org/SDL/blob/a3546e50697f2424a39f991148708c60903c553b/src/render/SDL_render.c#L2429

Reply:
Fixed here: https://github.com/fabiangreffrath/woof/pull/1847/commits/83d3e0af3bdadf6368f6a3cb50168efb63884ec5

Reply:
First case should not be modified, your original code + uniform scale in letterboxed cases is correct approach

Reply:
So it's all correct now? 6aabb3f
If not, post your diff or the entire `UpdateViewport` function here.

Reply:
Looks good, thanks